include/devicetree.h

```diff
@@ -1503,6 +1512,79 @@
 #define DT_INST_IRQ_HAS_NAME(inst, name) \
 	DT_IRQ_HAS_NAME(DT_DRV_INST(inst), name)
 
+/**
+ * @brief Invokes given macro for all child nodes of a parent.
+ *
+ * @param parent	parent node to iterate
+ * @param fun		macro to invoke
+ *
+ * Macro should be defined to take two parameters, where first will be
+ * the parent node, and second will be child node provided by the loop.
+ *
+ * Example:
+ *
+ *	#define PARENT DT_INST(0, fixed_partitions)
+ *	#define MKSTR(discard, a) #a,
+ *	const char *nodes[] {
+ *		DT_FOR_EACH_CHILD(PARENT, MKSTR)
+ *	};
+ *
+ * Would generate list of strings representing child nodes, partitions,
+ * of first instance of "fixed-partitions" compatible.
+ *
+ */
+#define DT_FOR_EACH_CHILD(parent, fun) DT_FOR_EACH_CHILD_(parent, fun)
```

Comment:
There is a reason to do this here instead of using FOR_EACH from util, you can do:
`DT_FOR_EACH_CHILD(something, FOR_EACH(sth else))`
but you can not do:
`FOR_EACH(something, FOR_EACH(sth else)`